src/Nethermind/Nethermind.JsonRpc/Modules/Eth/SimulateTxExecutor.cs

```diff
@@ -69,17 +72,28 @@ protected override SimulatePayload<TransactionWithSourceDetails> Prepare(Simulat
         return result;
     }
 
-    private static void UpdateTxType(TransactionForRpc callTransactionModel)
+    private static TransactionForRpc UpdateTxType(TransactionForRpc rpcTransaction)
     {
-        if (callTransactionModel.Type == TxType.Legacy)
+        // TODO: This is a bit messy since we're changing the transaction type
+        if (rpcTransaction is LegacyTransactionForRpc legacy)
         {
-            callTransactionModel.Type = TxType.EIP1559;
+            rpcTransaction = new EIP1559TransactionForRpc
+            {
+                Nonce = legacy.Nonce,
+                To = legacy.To,
+                From = legacy.From,
+                Gas = legacy.Gas,
+                Value = legacy.Value,
+                Input = legacy.Input,
+                GasPrice = legacy.GasPrice,
+                ChainId = legacy.ChainId,
+                V = legacy.V,
+                R = legacy.R,
+                S = legacy.S,
+            };
```

Comment:
We're kind of "upgrading" the transaction type. Previous code did not take into consideration `AccessList` transaction (was this intentional)?